### PR TITLE
add case index

### PR DESCRIPTION
**Pull request by LeungChiNan**
## Motivation and Context

add more case for index function

## Description

add more case for index function

## Checklist
**Before PR**:

- [x] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Review by NeosZhang — APPROVED** [review]

File: diopi_test/python/conformance/diopi_configs.py [hunk 1]
```diff
@@ -6462,28 +6612,28 @@
...
+                    "shape": ((6, 4, 5), (6, 4, 5), (16, 4, 4), (64, 4, 14, 14), (4, 4)),
+                    "dtype": [Dtype.float32, Dtype.float64, Dtype.float16, Dtype.int32,
+                              Dtype.int64, Dtype.uint8, Dtype.int8, Dtype.bool],
                 },
                 {
                     "ins": ['indices1'],
-                    "shape": ((16,), (64,)),
-                    "dtype": [Dtype.int64],
-                    "gen_fn": dict(fn=Genfunc.randint, low=0, high=4),
+                    "shape": ((6,), (6,), (16, 4), (64, 4), (4, 4)),
```
Comment (yangbofun) [inline diopi_test/python/conformance/diopi_configs.py:6628]: 这个默认全测true了，indices1这个tensor中的部分元素是true，部分是false的情况没有测到。

File: diopi_test/python/conformance/diopi_functions.py [hunk 2]
```diff
@@ -3177,7 +3177,8 @@ def copy_(input, other) -> Tensor:
 
 def gather(input, dim, index):
     assert isinstance(dim, int), "dim must be int"
-    assert len(input.size().data) == len(index.size().data), "input and index must have the same number of dimensions"
+    if len(input.size().data) > 1 and len(index.size().data) > 1:
```
Comment (yangbofun) [inline diopi_test/python/conformance/diopi_functions.py:3180]: 如果input.size的长度是0， index.size()的长度是1的话, 是否会报错？

Comment (LeungChiNan) [reply diopi_test/python/conformance/diopi_functions.py:3180]: 不会报错，且测例已包含
image
image

File: diopi_test/python/conformance/diopi_configs.py (hunk 1 above)
Comment (LeungChiNan) [reply diopi_test/python/conformance/diopi_configs.py:6628]: 此处测例是因为value的shape与indices中为True的个数相关，所以两处测例把indices全置为False或True，已添加通过value控制indices中True个数的测例
image

**Pull request merged**